Unity/Champloo/Assets/Scripts/Player/Movement States/OnWall.cs

```diff
@@ -4,51 +4,60 @@
 public class OnWall : MovementState
 {
     [SerializeField] private float maxFallSpeed = 10f;
-    [SerializeField]
-    private Vector2 wallJumpClimbForces;
-    [SerializeField]
-    private Vector2 wallJumpOffForces;
-    [SerializeField]
-    private Vector2 wallLeapForces;
-    [SerializeField]
     private float wallStickTime;
 
     [SerializeField]
+    private Vector2 wallJumpVelocity;
+
+
+    private Vector2 direction;
+
+    //Wallride variables being made accessable
+
+    public float WallStickTime { get { return wallStickTime; } }
+
+    [SerializeField]
     private PlayRandomSource wallSlideSound;
     [SerializeField]
     private PlayRandomSource jumpSound;
 
     private float timeToWallUnstick;
 
     private bool jumped = false;
+    protected override void Start()
+    {
+        base.Start();
+    }
+
 
     public override Vector3 ApplyFriction(Vector3 velocity)
     {
         int wallDirX = (controller.collisions.Left) ? -1 : 1;
         float moveX = player.InputPlayer.GetAxis("Move Horizontal");
-        velocity.x = moveX;
-        if (timeToWallUnstick > 0)
+        if (!jumped)
```

Comment:
nit: You might want to do the inverse of this check, and just return velocity instead. Then you can un-nest the next if statement